gammapy/maps/wcs/ndmap.py

```diff
@@ -297,7 +297,7 @@ def upsample(self, factor, order=0, preserve_counts=True, axis_name=None):
             ) + idx[2:]
         else:
             pix = list(idx)
-            idx_ax = self.geom.axes.index(axis_name)
+            idx_ax = 2 + self.geom.axes.index(axis_name)
```

Comment:
To avoid the hard-coded 2 here, maybe use:
```suggestion
            idx_ax = self.geom.axes_names.index(axis_name)
```